linode/py-metadata
# new: Add debug flag to view MDS API requests

## 📝 Description

This change allow users to enable debug mode and view logs from metadata service API request and response. User can specify `debug` flag and `debug_file` location to output the debug logs.

## ✔️ How to Test

1. In a development Linode, clone this repo and pull the change.
2. Run `make install`
3. Open `main.py`, create a MetadataClient and enable debug mode without specifying the output location:
```
from linode_metadata import MetadataClient

def main():
    client = MetadataClient(debug=True)
    # All of these responses are handled as DataClasses,
    # allowing IDEs to properly use completions.
    instance_info = client.get_instance()
    network_info = client.get_network()
    ssh_info = client.get_ssh_keys()
    user_data = client.get_user_data()
    print("Instance Info:", instance_info, "\n")
    print("Network Info:", network_info, "\n")
    print("SSH Keys:", "; ".join(ssh_info.users.root), "\n")
    print("User Data:", "\n", user_data)


if __name__ == "__main__":
    main()
```
4. Run the script and observe the debug output on the terminal window.
5. Specify a debug output file location when create the MetadataClient, for example:
```
    client = MetadataClient(debug=True, debug_file="output.txt")
```
6. Open `output.txt` and observe the debug output.
